core/extension/frontend_types.d.ts

```diff
@@ -0,0 +1,361 @@
+  GraphicShape2d,
+  GraphicSolidPrimitive,
+  HitListHolder,
+  IModelAppOptions,
+  IModelIdArg,
+  IpcAppOptions,
+  MarginOptions,
+  NativeAppOpts,
+  OffScreenViewportOptions,
+  OnViewExtentsError,
+  OsmBuildingDisplayOptions,
+  ParsedKeyin,
+  ParseKeyinError,
+  ParticleCollectionBuilder,
+  ParticleCollectionBuilderParams,
+  ParticleProps,
+  PickableGraphicOptions,
+  ScreenSpaceEffectBuilder,
+  ScreenSpaceEffectBuilderParams,
+  ScreenSpaceEffectContext,
+  ScreenSpaceEffectSource,
+  SelectAddEvent,
+  SelectedViewportChangedArgs,
+  SelectRemoveEvent,
+  SelectReplaceEvent,
+  TextureCacheOwnership,
+  TextureImage,
+  TileContent,
+  TiledGraphicsProvider,
+  TileDrawArgParams,
+  TileParams,
+  TileTreeDiscloser,
+  TileTreeOwner,
+  TileTreeParams,
+  TileTreeSupplier,
+  ToolAssistanceInstruction,
+  ToolAssistanceInstructions,
+  ToolAssistanceKeyboardInfo,
+  ToolAssistanceSection,
+  ToolTipOptions,
+  Uniform,
+  UniformArrayParams,
+  UniformContext,
+  UniformParams,
+  ViewAnimationOptions,
+  ViewChangeOptions,
+  ViewClipEventHandler,
+  ViewCreator2dOptions,
+  ViewCreator3dOptions,
+  ViewportDecorator,
+  ViewportGraphicBuilderOptions,
+  ZoomToOptions,
+} from "@itwin/core-frontend";
+
+// Types
+export type {
+  CanvasDecorationList,
+  DownloadBriefcaseId,
+  DownloadBriefcaseOptions,
+  FlashSettingsOptions,
+  GpuMemoryLimit,
+  GraphicList,
+  GraphicPrimitive,
+  MarkerFillStyle,
+  MarkerImage,
+  MarkerTextAlign,
+  MarkerTextBaseline,
+  OnFlashedIdChangedEventArgs,
+  ParseKeyinResult,
+  SelectionSetEvent,
+  TextureCacheKey,
+  TextureImageSource,
+  TextureOwnership,
+  ToolList,
+  ToolType,
+} from "@itwin/core-frontend";
+
+// Enums (note these are all real!)
+export {
+  ACSDisplayOptions,
+  ACSType,
+  ActivityMessageEndReason,
+  BeButton,
+  BeModifierKeys,
+  ClipEventType,
+  ContextRotationId,
+  CoordinateLockOverrides,
+  CoordSource,
+  CoordSystem,
+  DepthPointSource,
+  EventHandled,
+  FlashMode,
+  FrontendLoggerCategory,
+  GraphicType,
+  HitDetailType,
+  HitGeomType,
+  HitParentGeomType,
+  HitPriority,
+  HitSource,
+  InputSource,
+  KeyinParseError,
+  LocateAction,
+  LocateFilterStatus,
+  ManipulatorToolEvent,
+  MessageBoxIconType,
+  MessageBoxType,
+  MessageBoxValue,
+  OutputMessageAlert,
+  OutputMessagePriority,
+  OutputMessageType,
+  ParseAndRunResult,
+  SelectionMethod,
+  SelectionMode,
+  SelectionProcessing,
+  SelectionSetEventType,
+  SnapHeat,
+  SnapMode,
+  SnapStatus,
+  StandardViewId,
+  StartOrResume,
+  TextureTransparency,
+  TileBoundingBoxes,
+  TileGraphicType,
+  TileLoadPriority,
+  TileLoadStatus,
+  TileTreeLoadStatus,
+  TileVisibility,
+  ToolAssistanceImage,
+  ToolAssistanceInputMethod,
+  UniformType,
+  VaryingType,
+  ViewStatus,
+  ViewUndoEvent,
+} from "@itwin/core-frontend";
+
+// Abstract Classes
+export type {
+  AuxCoordSystemState,
+  BriefcaseNotificationHandler,
+  DisplayStyleState,
+  GeometricModelState,
```

Comment:
Has anyone determined how/if `instanceof` checks can be made to work when we're exporting only the types?

Reply:
What use cases are you interested in?
If you need to check the type of a variable that is passed in and was instantiated using the class's constructor then we can expose a function that effectively wraps the instanceof method.
However, if you plan to create the variable in the extension without the constructor then the best approach may require checking all required properties. This is far from ideal since it will likely be fragile, but I have not been able to find a better way to check type while hiding a class and its methods.

Reply:
```
function doSomethingWithAView(view: ViewState) {
  if (view instanceof DrawingViewState)
    // ...
}
```

Reply:
Is plan for the `view` parameter to be created on our side using the DrawingViewState constructor, or is it created in an extension as an object matching the DrawingViewState type?

Reply:
I am only asking about types that we don't expect the extension to instantiate - the only reason they would want a real export instead of a type export is to be able to use `instanceof`.

Reply:
I believe the best solution to this would be to access the object's constructor name using `obj.constructor.name` which would show the exact class used to instantiate the object. This would cover the use case you described where the object is passed in and you want to check if it is a specific subclass of that type.

Pros: lightweight, built-in without maintenance, universal
Cons: sensitive to minification, unable to identify superclasses, dependent on magic string comparisons

Reply:
That's not generally going to suffice. We'll have to revisit this sometime.